lib/spack/spack/bootstrap/clingo.py

```diff
@@ -119,14 +139,40 @@ def concretize(self) -> "spack.spec.Spec":
         s = spack.spec.Spec.from_specfile(str(self.prototype_path()))
         s._mark_concrete(False)
 
-        # Tweak it to conform to the host architecture
+        # These are nodes in the cmake stack, whose versions are frequently deprecated for
+        # security reasons. In case there is no external cmake on this machine, we'll update
+        # their versions to the most preferred, within the valid range, according to the
+        # repository we know.
+        to_be_updated = {
+            pkg_name: (spack.repo.PATH.get_pkg_class(pkg_name), valid_versions)
+            for pkg_name, valid_versions in {
+                "ca-certificates-mozilla": ":",
+                "openssl": "3:3",
+                "curl": "8:8",
+                "cmake": "3.16:3",
+                "libiconv": "1:1",
+                "ncurses": "6:6",
+                "m4": "1.4",
+            }.items()
+        }
+
+        # Tweak it to conform to the host architecture + update the version of a few dependencies
         for node in s.traverse():
+            # Clear patches, we'll compute them correctly later
+            node.patches.clear()
+            if "patches" in node.variants:
+                del node.variants["patches"]
```

Comment:
@alalazo should we make this only clear the patches when we actually modify the node?

Reply:
Nope. We try to apply patches by iterating over variants iirc, so we need to be consistent and re-do that post solve step.